server/middlewares/authenticationMiddleware.ts

```diff
@@ -1,63 +1,24 @@
 import { Request, Response, NextFunction } from 'express';
 import * as jwt from 'jsonwebtoken';
-// import { verifyAzureToken } from 'azure-ad-jwt-lite';
-// import jwksClient from 'jwks-rsa';
+import jwksClient from 'jwks-rsa';
+import catchAsync from '../utils/catchAsync';
 
-export default async (req: Request, res: Response, next: NextFunction) => {
-  let token = '';
-
-  if (req.cookies.jwt) {
-    token = req.cookies.jwt;
-  } else {
-    return res.status(401).json({
-      message: 'Authentication failed',
-    });
-  }
-
-  // TU JEST NAJLEPSZY SPOSOB ALE NO AAAAAAAAA MS SPAPRAŁ ROBOTE
-  // const decoded = await verifyAzureToken(token, {
-  //   issuer: `https://sts.windows.net/${process.env.TENANT_ID}/`,
-  //   audience: `api://${process.env.CLIENT_ID}/user.read`,
-  // }) as any;
-  // OPCJA NR 2 lub 1
-  // const decoded = await verifyAzureToken(token) as any;
-
-  // req.decoded = {
-  //   userMSId: decoded.tid as string,
-  // };
-  // next();
-
-  // TUTAJ również poprawna bardziej basic opcja
-  /* const client = jwksClient({
-    jwksUri: `https://login.microsoftonline.com/${process.env.TENANT_ID}/discovery/keys`,
-    // jwksUri: `https://login.microsoftonline.com/${process.env.TENANT_ID}/discovery/v2.0/keys`,
+export default catchAsync(async (req: Request, res: Response, next: NextFunction) => {
+  const client = jwksClient({
+    jwksUri: `https://login.microsoftonline.com/${process.env.TENANT_ID}/discovery/v2.0/keys`,
   });
-  function getKey(header: jwt.JwtHeader, callback: jwt.SigningKeyCallback) {
-    client.getSigningKey(header.kid, (err, key) => {
-      if (err) {
-        return res.status(401).json({
-          message: 'Authentication failed',
-        });
-      }
-
-      const signingKey = key.getPublicKey();
-      console.log('signingKey', signingKey);
-      callback(err, signingKey);
-    });
-  }
-*/
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  const decoded = jwt.decode(token) as any;
-
-  if (!decoded) {
+  const token = req.cookies.validation_token;
+  const decoded: any = jwt.decode(token, { complete: true });
+  const key = await client.getSigningKey(decoded?.header.kid);
+  const validToken: any = jwt.verify(token, key.getPublicKey());
+  if (validToken) {
+    req.decoded = {
+      userMSId: validToken.oid,
```

Comment:
We could rename this to `userId` because we store MSId as normal id in db

Reply:
I think `userMSId` is reffered in several places. I would have to change this also there. I will check this today.